qcelemental/tests/test_molecule.py

```diff
@@ -190,13 +235,27 @@ def test_water_orient():
     assert frag_0_1.get_hash() != frag_1_0.get_hash()
 
 
-def test_molecule_errors():
+def test_molecule_errors_extra():
     data = water_dimer_minima.dict()
     data["whatever"] = 5
     with pytest.raises(ValidationError):
         Molecule(**data)
 
 
+def test_molecule_errors_connectivity():
+    data = water_molecule.dict()
+    data["connectivity"] = [(-1, 5, 5)]
+    with pytest.raises(ValidationError):
+        Molecule(**data)
+
+
+def test_molecule_errors_shape():
+    data = water_molecule.dict()
+    data["geometry"] = list(range(8))
+    with pytest.raises(ValidationError):
+        Molecule(**data)
```

Comment:
For these very generic errors, if there’s more specific info in the error msg, it’s sometimes handy to test the string contents too.

Reply:
Agreed, best to put it on a TODO.